rpm/mir.spec

```diff
@@ -0,0 +1,608 @@
+BuildRequires:  pkgconfig(wayland-client)
+BuildRequires:  pkgconfig(xcb)
+BuildRequires:  pkgconfig(xcb-composite)
+BuildRequires:  pkgconfig(xcb-xfixes)
+BuildRequires:  pkgconfig(xcb-render)
+BuildRequires:  pkgconfig(xcursor)
+BuildRequires:  pkgconfig(xkbcommon)
+BuildRequires:  pkgconfig(xkbcommon-x11)
+BuildRequires:  pkgconfig(yaml-cpp)
+BuildRequires:  pkgconfig(wlcs)
+
+# pkgconfig(egl) is now from glvnd, so we need to manually pull this in for the Mesa specific bits...
+BuildRequires:  mesa-libEGL-devel
+
+# For some reason, this doesn't get pulled in automatically into the buildroot
+BuildRequires:  libatomic
+
+# For detecting the font for CMake
+BuildRequires:  gnu-free-sans-fonts
+
+# For validating the desktop file for mir-demos
+BuildRequires:  %{_bindir}/desktop-file-validate
+
+# Add architectures as verified to work
+%ifarch %{ix86} x86_64 %{arm} aarch64
+BuildRequires:  valgrind
+%endif
+
+
+%description
+Mir is a Wayland display server toolkit for Linux systems,
+with a focus on efficiency, robust operation,
+and a well-defined driver model.
+
+%package devel
+Summary:       Development files for Mir
+Requires:      %{name}-common-libs%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:      %{name}-server-libs%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:      %{name}-lomiri-libs%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:      %{name}-test-libs-static%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+# Documentation can no longer be built properly
+Obsoletes:     %{name}-doc < 2.15.0
+
+%description devel
+This package provides the development files to create compositors
+built on Mir.
+
+%package internal-devel
+Summary:       Development files for Mir exposing private internals
+Requires:      %{name}-devel%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+
+%description internal-devel
+This package provides extra development files to create compositors
+built on Mir that need access to private internal interfaces.
+
+%package common-libs
+Summary:       Common libraries for Mir
+License:       LGPL-2.1-only or LGPL-3.0-only
+# mirclient is gone...
+Obsoletes:     %{name}-client-libs < 2.6.0
+# debug extension for mirclient is gone...
+Obsoletes:     %{name}-client-libs-debugext < 1.6.0
+# mir utils are gone...
+Obsoletes:     %{name}-utils < 2.0.0
+# Ensure older mirclient doesn't mix in
+Conflicts:     %{name}-client-libs < 2.6.0
+
+%description common-libs
+This package provides the libraries common to be used
+by Mir clients or Mir servers.
+
+%package lomiri-libs
+Summary:       Lomiri compatibility libraries for Mir
+License:       GPL-2.0-only or GPL-3.0-only
+Requires:      %{name}-common-libs%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:      %{name}-server-libs%{?_isa} = %{?epoch:%{epoch}:}%{version}-%{release}
+
+%description lomiri-libs
+This package provides the libraries for Lomiri to use Mir
+as a Wayland compositor.
```

Comment:
miroil isn't, necessarily, lomiri-only, should we encode that here?

Reply:
What is miroil for then?

Reply:
Yeah, so I'll leave this then.